openhands/events/stream.py

```diff
@@ -219,10 +219,26 @@ def set_secrets(self, secrets: dict[str, str]) -> None:
     def update_secrets(self, secrets: dict[str, str]) -> None:
         self.secrets.update(secrets)
 
-    def _replace_secrets(self, data: dict[str, Any]) -> dict[str, Any]:
+    def _replace_secrets(
+        self, data: dict[str, Any], is_top_level: bool = True
+    ) -> dict[str, Any]:
+        # Fields that should not have secrets replaced (only at top level - system metadata)
+        TOP_LEVEL_PROTECTED_FIELDS = {
+            'timestamp',
+            'id',
+            'source',
+            'cause',
+            'action',
+            'observation',
```

Comment:
NIT: in the future we should rename this to `observation_type`; at first I thought this was the observation contents but we have a separate field `content` for all observation classes

Reply:
yeah we should maybe just do "event_type":"action:XXX" :( but it is probably a big change that needs some backward compatibility support..